PunchCard.py

```diff
@@ -27,10 +28,18 @@ def calcWorkTime(timeIn, timeOut):
 def calculateDay(dayEntry):
     index = 0
     dayHours = 0.0
+    errors = ''
     while(index < len(dayEntry)):
+        if not validTime(dayEntry[index]) or not validTime(dayEntry[index+1]):
+            if not validTime(dayEntry[index]):
+                errors = '{}\nInvalid time: {}'.format(errors, dayEntry[index])
+            if not validTime(dayEntry[index+1]):
```

Comment:
Ah, I see what you are doing. How about making errors an array instead and pushing errors on to it? You could join them below on line 42, or return the array and handle it as an array.